src/main/java/com/redhat/devtools/lsp4ij/features/codeBlockProvider/LSPCodeBlockProvider.java

```diff
@@ -78,50 +92,65 @@ public TextRange getCodeBlockRange(Editor editor, PsiFile file) {
                 closeBraceChar,
                 closeBraceOffset
         );
-        if (codeBlockRange != null) {
-            return codeBlockRange;
-        }
 
         // Failing that, try to find it using the folding ranges
-        return getUsingFoldingRanges(
-                file,
-                document,
-                offset,
-                openBraceChar,
-                openBraceOffset,
-                closeBraceChar,
-                closeBraceOffset
-        );
+        if (codeBlockRange == null) {
```

Comment:
Make this more robust so that if a code block text range anchored by a brace pair character does not yield a code block, it will search for a code block as if not anchored.